packages/react-ui-kit/src/Form/Select.tsx

```diff
@@ -242,13 +247,22 @@ export const Select = ({
     };
   }, []);
 
+  useEffect(() => {
+    if (value) {
+      const valueIdx = options.findIndex(option => option.value === value.value);
+      setSelectedOption(valueIdx);
+    }
+  }, [options, value]);
```

Comment:
You don't need a `useEffect` here. You are just creating new state from existing state, there is no effect there